status/status.go

```diff
@@ -73,9 +73,11 @@ func FromProto(s *spb.Status) *Status {
 	return status.FromProto(s)
 }
 
-// FromError returns a Status representing err if it was produced from this
-// package or has a method `GRPCStatus() *Status`. Otherwise, ok is false and a
-// Status is returned with codes.Unknown and the original error message.
+// FromError returns a Status representing err.
+// If err is nil, ok is true and a Status with codes.OK is returned.
+// If it has a method `GRPCStatus() *Status`, ok is true and this value is returned.
```

Comment:
`Errorf()`'s return values have the method `GRPCStatus() *Status`; however, this is not documented anywhere.  It should be explicit that this function turns `error`s produced by `Errorf` or `(*Status).Err()` into `*Status`es, as that is its primary function.

Why not something more like the original text?
```go
// FromError returns a Status representing err if it was produced by this
// package or has a method `GRPCStatus() *Status`.
// If err is nil, a Status is returned with codes.OK and no message.
// Otherwise, ok is false and a Status is returned with codes.Unknown and
// the original error message.
```

Reply:
In the end it's up to you. My goal by changing that was to make it very explicit that we return `ok=true` for `err==nil` because that can be against intuition if you are under the impression this works like a type assertion.

Take the following code fragment as example:
```go
func doSomething() error {
  return nil
}
// ...
err := doSomething()
if err, ok := err.(*SpecificType); ok {
   // common style to check for error, see https://blog.golang.org/go1.13-errors#TOC_2.1.
}

if status, ok := status.FromError(err); ok {
   // oh status is actually OK so no error happened. So this code is wrong.
   // The writer was under the wrong impression that this works like the case above
}
```

Feel free to change this PR if you have an idea about how to improve it better.

Reply:
> My goal by changing that was to make it very explicit that we return ok=true for err==nil  

The text I proposed above seems pretty explicit that `ok=true` when `err=nil`, as that case is documented above the "Otherwise, ok is false" sentence.

Your example code is highly unusual.  Why would you test `err` to see what type it is before checking to see if there even *was* an error by comparing with `nil`?

Reply:
It seems your judgement about this issue is different from mine. I really don't mind and accept that and will close this PR. There are probably more important fixes waiting than this.
Feel free to close the linked issue and tag it with "Wontfix".